docs/CHANGELOG.rst

```diff
@@ -32,12 +32,12 @@ Main new features
 - Run time display color format adjustment with RGB888 support
 - Built-in support ``pthread``, ``FreeRTOS`` and other (RT)OSes which are used during rendering
 - Built-in support LVGL's, C library, and other stdlib's
-- Better parallel rendering architecture. See the details :ref:`here <porting/draw/custom gpu [0-1]>`
+- Better parallel rendering architecture. See the details :ref:`here <porting/draw/custom gpu>`
 - Built in display and touch driver: SDL, Linux Frame buffer,  NuttX LCD and touch drivers, ST7789 and ILI9341 driver are available and more will come soon
-- :ref:`others/observer/observer [0-1]` allows to bind data to UI elements and create a uniform and easy to maintain API
+- :ref:`others/observer/index/observer` allows to bind data to UI elements and create a uniform and easy to maintain API
 - GitHub CodeSpace integration makes possible to run LVGL in an Online VSCode editor with 3 click. See more `here <https://blog.lvgl.io/2023-04-13/monthly-newsletter>`__
 - Add vector graphics support via ThorVG. It can be used to draw vector graphics to a `Canvas <https://github.com/lvgl/lvgl/blob/master/examples/widgets/canvas/lv_example_canvas_8.c>`__
-- :ref:`widgets/image/image (lv_image) [0-1]` supports aligning, stretching or tiling the image source if the widget is larger or smaller.
+- :ref:`widgets/image/image (lv_image)` supports aligning, stretching or tiling the image source if the widget is larger or smaller.
```

Comment:
What about using the string from the URL as it is? It would make it simple to just copy paste it.
```suggestion
- :ref:`widgets/image.html#overview` supports aligning, stretching or tiling the image source if the widget is larger or smaller.
```

Reply:
I don't understand what you are saying... 

this 

`widgets/image.html#overview` is an invalid reference. I am not even able to define it because the # is an illegal character. Well not an an easy way anyhow, I would have to write a special handler for sphinx in order to create those kinds of references and they could not be defined manually only programmatically. What I mean by that is I would not be able to use `.. _some_ref#some section:` in the rst files. How the references are currently done is the documents folder gets copied to a temp directory and the script edits those files and adds adds the reference declarations as if you had actually keyed them in yourself. It is nice this way because I am able to easily dump a list of the references or go into the temp folder and look at what it is doing to see if there are any problems. Hiding everything behind the scenes so to speak makes it really hard to locate where problems are coming from.

Reply:
What about this idea?

Reply:
It's a pity that `#` is illegal. What about some other separator between the page and  the section? E.g. `this_is/a_page__section`